src/main/kotlin/io/github/pavleprica/kotlin/cache/Cache.kt

```diff
@@ -9,7 +9,7 @@ import java.util.*
  * @param E is the value of the cache.
  * @property size is the current size of the cache.
  */
-interface Cache<T, E> {
+interface Cache<T: Any, E: Any> {
```

Comment:
Interesting, is this a requirement of 1.6 or?

Reply:
On 2nd consideration it may be a good idea to allow keys with value `null`, but the reason for this change is that there was an warning from the compiler that without an upper bound of `Any`, the constructor for `Optional` could potentially be passed a NULL value which would violate its constraints.
This would become a compilation-error in future.

So I thought it would be good to apply this change throughout the code, at least for the cache-values.

Reply:
Makes sense